velox/functions/prestosql/tests/BinaryFunctionsTest.cpp

```diff
@@ -747,4 +747,65 @@ TEST_F(BinaryFunctionsTest, fromIEEE754Bits32) {
       fromIEEE754Bits32(hexToDec("0000000000000001")),
       "Input floating-point value must be exactly 4 bytes long");
 }
+
+TEST_F(BinaryFunctionsTest, rpad) {
+  const auto rpad = [&](std::optional<StringView> arg,
+                        std::optional<StringView> key,
+                        std::optional<int32_t> size) {
+    return evaluateOnce<std::string>(
+        "rpad(c0, c1, c2)",
+        makeRowVector(
+            {vectorMaker_.flatVectorNullable<StringView>({arg}, VARBINARY()),
+             vectorMaker_.flatVectorNullable<StringView>({key}, VARBINARY()),
+             vectorMaker_.flatVectorNullable<int32_t>({size}, INTEGER())}));
+  };
+  EXPECT_EQ(rpad("1234", "45", 14), ("12344545454545"));
+  EXPECT_EQ(
+      rpad("123456789012", "4444555566667777", 24),
+      ("123456789012444455556666"));
+  EXPECT_EQ(
+      rpad("1234567890", "44445555666677778888", 30),
+      ("123456789044445555666677778888"));
+  EXPECT_EQ(rpad("1234", "45", 15), ("123445454545454"));
+  EXPECT_EQ(rpad("1234", "4524", 14), ("12344524452445"));
+  EXPECT_EQ(rpad("1234", "4524", 6), ("123445"));
+  EXPECT_EQ(rpad("23", "4524", 0), (""));
+  EXPECT_EQ(rpad("∆", "˚", 2), ("∆˚"));
+  EXPECT_EQ(rpad("©", "£", 4), ("©£££"));
+  EXPECT_EQ(rpad("1234", "4524", 2), ("12"));
+  EXPECT_EQ(rpad(std::nullopt, std::nullopt, 0), std::nullopt);
+  EXPECT_THROW(rpad("2312", "4524", -1), VeloxUserError);
+  EXPECT_THROW(rpad("2312", "", 1), VeloxUserError);
+}
+
+TEST_F(BinaryFunctionsTest, lpad) {
+  const auto lpad = [&](std::optional<StringView> arg,
+                        std::optional<StringView> key,
+                        std::optional<int32_t> size) {
+    return evaluateOnce<std::string>(
+        "lpad(c0, c1, c2)",
+        makeRowVector(
+            {vectorMaker_.flatVectorNullable<StringView>({arg}, VARBINARY()),
+             vectorMaker_.flatVectorNullable<StringView>({key}, VARBINARY()),
+             vectorMaker_.flatVectorNullable<int32_t>({size}, INTEGER())}));
+  };
+  EXPECT_EQ(lpad("1234", "45", 14), ("45454545451234"));
+  EXPECT_EQ(lpad("1234", "45", 15), ("454545454541234"));
+  EXPECT_EQ(
+      lpad("123456789012", "4444555566667777", 24),
+      ("444455556666123456789012"));
+  EXPECT_EQ(
+      lpad("1234567890", "44445555666677778888", 30),
+      ("444455556666777788881234567890"));
+  EXPECT_EQ(lpad("∆", "˚", 2), ("˚∆"));
+  EXPECT_EQ(lpad("©", "£", 4), ("£££©"));
+  EXPECT_EQ(lpad("1234", "4524", 14), ("45244524451234"));
+  EXPECT_EQ(lpad("1234", "4524", 6), ("451234"));
+  EXPECT_EQ(lpad("1234", "4524", 0), (""));
+  EXPECT_EQ(lpad("1234", "4524", 2), ("12"));
```

Comment:
This is an interesting use case. Somehow from the docs I was expecting that rpad truncates characters from the right and lpad truncates from the left. Is this not the case? Might be worth clarifying.

Reply:
i believe that this is expected behavior... in StringFunctionsTest.cpp I notice the same kind of behavior:
 EXPECT_EQ("tex", lpad("text", 3, "xy"));
which is the same as
  EXPECT_EQ("tex", rpad("text", 3, "xy"));
  
Also, is this behavior something you'd prefer i document in binary.rst in the comments for clarity?

Reply:
Got it. Thank you for clarifying. It would be nice to document this in binary.rst as it is not obvious.

Reply:
just made the change, please take a look